src/MongoDB.Driver/Linq/Translators/PredicateTranslator.cs

```diff
@@ -1653,6 +1655,16 @@ private IFieldExpression GetFieldExpression(Expression expression)
             return fieldExpression;
         }
 
+        private void ValidatePipelineExpressionThrowIfNotValid(WhereExpression whereExpression)
+        {
+            var memberInfoCollection = OutOfCurrentScopePrefixCollector.Collect(whereExpression)?.ToList();
+            var unsupportedField = memberInfoCollection?.FirstOrDefault(c => !string.IsNullOrWhiteSpace(c.OutOfCurrentScopePrefix));
+            if (unsupportedField != null)
+            {
+                throw new NotSupportedException($"The LINQ expression: {whereExpression} has the member \"{unsupportedField.OutOfCurrentScopePrefix}\" which can not be used to build a correct Mongo query.");
```

Comment:
I think we're always supposed to use MongoDB instead of "Mongo".

Reply:
Done